Signum.React.Extensions/Authorization/AuthController.cs

```diff
@@ -33,13 +33,15 @@ public LoginResponse Login([FromBody]LoginRequest data)
                 else
                     user = AuthLogic.Authorizer.Login(data.userName, data.password);
             }
-            catch (Exception e) when (e is IncorrectUsernameException || e is IncorrectPasswordException)
+            catch (Exception e) when (e is IncorrectUsernameException || e is IncorrectPasswordException || e is AuthenticationFailedException)
             {
-                if (AuthServer.MergeInvalidUsernameAndPasswordMessages)
+                if (e is AuthenticationFailedException)
                 {
-                    ModelState.AddModelError("userName", AuthMessage.InvalidUsernameOrPassword.NiceToString());
-                    ModelState.AddModelError("password", AuthMessage.InvalidUsernameOrPassword.NiceToString());
-                    throw new HttpResponseException(Request.CreateErrorResponse(HttpStatusCode.BadRequest, this.ModelState));
+                    throw ModelException("login", e.Message);
+                }
+                else if (AuthServer.MergeInvalidUsernameAndPasswordMessages)
+                {
+                    throw ModelException("login", AuthMessage.InvalidUsernameOrPassword.NiceToString());
                 }

```

Comment:
Why you removed ModelState.AddModelError?

Reply:
Because when we have AuthServer.MergeInvalidUsernameAndPasswordMessages then there is no exact control to show validation red border. I prefer to show the message under Login button instead of duplicating same merged error message for both username and password controls.